### PR TITLE
snobol4 2.3.1

### PR DESCRIPTION
**Pull request by philbudne**
Updated snobol4 formula from version 1.5 (c. 2013).
I'm the source package creator/maintainer.

### CODE REVIEW

File: Formula/snobol4.rb [hunk 1]
```diff
@@ -1,9 +1,8 @@
 class Snobol4 < Formula
   desc "String oriented and symbolic programming language"
-  homepage "https://www.snobol4.org/"
-  url "ftp://ftp.ultimate.com/snobol/old/snobol4-1.5.tar.gz"
-  mirror "https://src.fedoraproject.org/lookaside/pkgs/snobol/snobol4-1.5.tar.gz/54ac3ddd51fb34ec63b1eb0ae7b99794/snobol4-1.5.tar.gz"
-  sha256 "9f7ec649f2d700a30091af3bbd68db90b916d728200f915b1ba522bcfd0d7abd"
+  homepage "https://www.regressive.org/snobol4/"
+  url "https://ftp.regressive.org/snobol/snobol4-2.3.1.tar.gz"
+  sha256 "91244d67d4e29d2aadce5655bd4382ffab44c624a7ea4ad6411427f3abf17535"
```
Comment (SMillerDev) [inline Formula/snobol4.rb:5]: Is there any documentation somewhere of the hosting change?

File: Formula/snobol4.rb [hunk 2]
```diff
@@ -19,8 +18,58 @@ class Snobol4 < Formula
...
+  depends_on "libffi"
+  depends_on "openssl@1.1"
+  depends_on "readline"
+  depends_on "sqlite"
+  depends_on "zlib"
+
   def install
     system "./configure", "--prefix=#{prefix}", "--mandir=#{man}"
-    system "make", "install"
+    system "make", "-j1", "install_notiming"
```
Comment (SMillerDev) [inline Formula/snobol4.rb:30]:
```suggestion
    ENV.deparallelize
    system "make", "install_notiming"
```

File: Formula/snobol4.rb [hunk 3]
```diff
@@ -19,8 +18,58 @@ class Snobol4 < Formula
...
+        ident(v, v2)                    :f(fail)
+
+* more tests here? (set t to test name before each test)
+	output = 'OK'
+        :(end)
+
+fail output = t ' test failed at ' &LASTLINE
+        &code = 1
+end
+")
```
Comment (SMillerDev) [inline Formula/snobol4.rb:72]:
```suggestion
    testfile.write <<~EOS
      -include 'digest.sno'
      -include 'dirs.sno'
      -include 'ezio.sno'
      -include 'ffi.sno'
      -include 'fork.sno'
      -include 'ndbm.sno'
      -include 'logic.sno'
      -include 'random.sno'
      -include 'sprintf.sno'
      -include 'sqlite3.sno'
      -include 'stat.sno'
      -include 'zlib.sno'

      * DBM test
              t = 'dbm'
              k = '🍺'
              v = '🙂'
              fn = '#{dbmfile}'
              h1 = dbm_open(fn, 'cw')         :f(fail)
              dbm_store(h1, k, v)             :f(fail)
              dbm_close(h1)                   :f(fail)
              h2 = dbm_open(fn, 'r')          :f(fail)
              v2 = dbm_fetch(h2, k)           :f(fail)
              ident(v, v2)                    :f(fail)

      * more tests here? (set t to test name before each test)
	      output = 'OK'
              :(end)

      fail output = t ' test failed at ' &LASTLINE
              &code = 1
      end
    EOS
```

Comment (SMillerDev) [inline Formula/snobol4.rb:72]:
```suggestion
    testfile.write <<~EOS
      -include 'digest.sno'
      -include 'dirs.sno'
      -include 'ezio.sno'
      -include 'ffi.sno'
      -include 'fork.sno'
      -include 'ndbm.sno'
      -include 'logic.sno'
      -include 'random.sno'
      -include 'sprintf.sno'
      -include 'sqlite3.sno'
      -include 'stat.sno'
      -include 'zlib.sno'

      * DBM test
              t = 'dbm'
              k = '🍺'
              v = '🙂'
              fn = '#{dbmfile}'
              h1 = dbm_open(fn, 'cw')         :f(fail)
              dbm_store(h1, k, v)             :f(fail)
              dbm_close(h1)                   :f(fail)
              h2 = dbm_open(fn, 'r')          :f(fail)
              v2 = dbm_fetch(h2, k)           :f(fail)
              ident(v, v2)                    :f(fail)

      * more tests here? (set t to test name before each test)
	      output = 'OK'
              :(end)

      fail output = t ' test failed at ' &LASTLINE
              &code = 1
      end
    EOS
```

Comment (SMillerDev) [inline Formula/snobol4.rb:72]:
```suggestion
    (testpath/"test.sno").write <<~EOS
      -include 'digest.sno'
      -include 'dirs.sno'
      -include 'ezio.sno'
      -include 'ffi.sno'
      -include 'fork.sno'
      -include 'ndbm.sno'
      -include 'logic.sno'
      -include 'random.sno'
      -include 'sprintf.sno'
      -include 'sqlite3.sno'
      -include 'stat.sno'
      -include 'zlib.sno'

      * DBM test
              t = 'dbm'
              k = '🍺'
              v = '🙂'
              fn = '#{dbmfile}'
              h1 = dbm_open(fn, 'cw')         :f(fail)
              dbm_store(h1, k, v)             :f(fail)
              dbm_close(h1)                   :f(fail)
              h2 = dbm_open(fn, 'r')          :f(fail)
              v2 = dbm_fetch(h2, k)           :f(fail)
              ident(v, v2)                    :f(fail)

      * more tests here? (set t to test name before each test)
	      output = 'OK'
              :(end)

      fail output = t ' test failed at ' &LASTLINE
              &code = 1
      end
    EOS
```

File: Formula/snobol4.rb (hunk 1 above)
Comment (philbudne) [reply Formula/snobol4.rb:6]: There's a one line mention in the distribution History (change log) file:
3/25/2022:
        LOTS: switch to regressive.org for web & mail

Comment (SMillerDev) [reply Formula/snobol4.rb:5]: But that's on this new website right? So that's not really reliable for this.

Comment (philbudne) [reply Formula/snobol4.rb:5]: Correct.  It wasn't a planned transition.  One of the links on wikipedia has been fixed, but there are a few broken ones left :-/

Comment (SMillerDev) [reply Formula/snobol4.rb:5]: Is there any external source that already verified the change?

**Comment by philbudne** [comment]
I announced the location change in ***@***.***,
but it doesn't look like the archive is public?

I updated the links, and put a note at

http://www.ultimate.com/phil/src.html#snobol

(which has a long history at web.archive.org, but not long enough to
pre-date the move from ultimate.com to snobol4.org (RIP))

File: Formula/snobol4.rb [hunk 4]
```diff
@@ -19,8 +18,58 @@ class Snobol4 < Formula
     sha256 yosemite:       "6903e1b05a795eae13f2f97fd2f1f4b883b03e1c94ba28e3747b3df98c6a955d"
   end
 
+  depends_on "libffi"
+  depends_on "openssl@1.1"
+  depends_on "readline"
+  depends_on "sqlite"
+  depends_on "zlib"
```
Comment (Bo98) [inline Formula/snobol4.rb:25]: Do these (ignoring OpenSSL) need the latest version or can the macOS version work? Apple's libffi for one is probably better than ours when it comes to arm64 trampoline handling.

Comment (Bo98) [reply Formula/snobol4.rb:25]: If the answer is "not sure", that's fine - I can test it out locally.

**Comment by philbudne** [comment]
> If the answer is "not sure", that's fine - I can test it out locally.

Not sure.

**Review by Bo98 — APPROVED** [review]
Thanks!

**Pull request closed without merging**